File: old/testerum-backend/testerum-runner/testerum-runner-api/src/main/kotlin/com/testerum/runner/events/model/StepStartEvent.kt
package com.testerum.runner.events.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.testerum.model.step.StepCall
import java.time.LocalDateTime

data class StepStartEvent @JsonCreator constructor(
        @JsonProperty("time")     override val time: LocalDateTime = LocalDateTime.now(),
        @JsonProperty("eventKey") override val eventKey: String,
        @JsonProperty("stepCall") val stepCall: StepCall
): RunnerEvent

File: old/web/web-frontend/src/app/model/test/event/step-start.event.ts
import {RunnerEvent} from "./runner.event";
import {StepCall} from "../../step/step-call.model";
import {RunnerEventTypeEnum} from "./enums/runner-event-type.enum";
import {Serializable} from "../../infrastructure/serializable.model";

export class StepStartEvent implements RunnerEvent, Serializable<StepStartEvent> {
    time: Date;
    eventKey: string;
    eventType: RunnerEventTypeEnum = RunnerEventTypeEnum.STEP_START_EVENT;

    stepCall: StepCall;

    deserialize(input: Object): StepStartEvent {
        this.eventKey = input["eventKey"];
        this.time = new Date(input["time"]);
        this.stepCall = new StepCall().deserialize(input["stepCall"]);
        return this;
    }

    serialize(): string {
        return undefined;
    }
}
